staging/src/k8s.io/apiserver/pkg/server/serve.go

```diff
@@ -115,6 +115,11 @@ func RunServer(server *http.Server, network string, stopCh <-chan struct{}) (int
 	go func() {
 		<-stopCh
 		ln.Close()
```

Comment:
this only mean that the listener is closed. it does not mean that no in-fly requests are handled.

Reply:
yeah, I agree. This way may still lose events, but it is the simplest way to reduce the events lost.

Reply:
>  but it is the simplest way to reduce the events lost

Right now you're waiting for `stopCh` to close and then closing `auditStopCh` that's passed to audit backend. The difference between this approach and the one that's currently in place is the time spent on calling `ln.Close()` which I believe is negligible